rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,225 @@
+
+For determining train location for the purposes of performing trip matching, existing data from RTR in the form of direction ID and next stop ID for trains will suffice for the time being.
+
+## Additional RTR prediction considerations
+
+Much like other realtime prediction systems, RTR generates predictions not only for currently-operating trips but for trips that have not yet begun where we are already tracking a vehicle on an earlier trip in the same block. This has posed problems with light rail as those operations are generally less block-based due to factors like having many yards at terminal locations allowing for more ad-hoc swapping of vehicles between trips. When inspectors enter car numbers on future trips this can, however, create implicit block relationships in realtime. This RFC does not directly solve the issue of how RTR should link subsequent trips for prediction purposes. What it does do, however, is make the relationship between realtime and schedule data clearer via better trip assignments, so that RTR can (for example) more easily factor in runs for the purposes of trip linking based on TODS data, if that is deemed to be a desirable approach.
+
+One important subtlety to note is that we will be relying on next stop ID as an input for Glides trip matching, but in order to determine this in the first place RTR needs to know what pattern a train is on in order to predict what stop it will visit next based on its predicted routing through the system. This raises an interesting question as to how, if at all, RTR should factor in the assigned trip provided by Glides when determining which pattern to assign to a train. If RTR does use the assigned trip for this purpose, there is at least a risk of a circular relationship between the two data points. Whatever the exact relationship ends up being, RTR should probably not take the trip as the only source of input for making this decision, as there are some known cases like short turning trains in the central subway where we know that a train may update its AVI on short notice and, currently at least, the Glides trainsheets may not necessarily be updated to reflect this change in a timely manner. Exactly where to find the balance between these factors is outside of the scope of this RFC, but it should be noted as a risk factor.
```

Comment:
Following up on [a recent discussion](https://github.com/mbta/technology-docs/pull/25/files#r1711532462 ), is this basically saying that RTR would still be expected to make a "preliminary" trip-trip assignment (for the purpose of knowing where it's going/what its next stop will be)? I guess the question of who/what determines a *trip assignment* need not be fully tied to who/what determines a vehicle's *predicted station stops*, as the latter could be influenced not only by trainsheet input but also the AVI codes and GTFS-alerts (and someone/thing needs to decide what how they get prioritized).

Reply:
Yes, one of my goals is to at least separate these concerns out a bit more explicitly. As I see it, what trips a train / crew will operate next is a bit more of a predictions issue and something we could probably mainly figure out with things like scheduled block / run data (pending validation of course). That said, as we brainstorm and validate approaches to predictions, I think we should think about whether there's an additional data that Glides could provide in order to make that determination easier - so I don't want to just drop the problem entirely on RTR, but I do want to tease it apart as a separate piece of logic.